cozmo_echo.py

```diff
@@ -7,30 +7,35 @@
 
 def main():
 
-    
+    start_dialogue = False
+
     my_convo = Dialogue('speech.wav')
     my_convo.get_cozmo_response("Hello! I will echo back what you say!")
-
-    try:
-        print("starting recording process")
-        recorder = Recorder("speech.wav")
-        print("Please say something nice into the microphone\n")
-        recorder.save_to_file()
-        print("Transcribing audio....\n")
+    
+    while not start_dialogue:
 
         try:
-            speech_text = my_convo.transcribe_audio()
-            my_convo.get_cozmo_response(speech_text)
+            print("starting recording process")
+            recorder = Recorder("speech.wav")
+            print("Please say something nice into the microphone\n")
+            recorder.save_to_file()
+            print("Transcribing audio....\n")
 
-        except:
-            traceback.print_exc()
-            print("error in speech detection")
-            my_convo.get_cozmo_response("Sorry I couldn't understand you")
-            sys.exit(0)
+            try:
+                speech_text = my_convo.transcribe_audio()
+                my_convo.get_cozmo_response(speech_text)
+                if "goodbye" or "bye" in speech_text:
```

Comment:
I think you meant `if "goodbye" in speech_text or "bye" in speech_text:` no?

What you have would not work as "goodbye" by itself is always true:

```
>>> if "goodbye" or "bye" in speech_text:
...     print("end")
...
end
>>> speech_text = "some text"
>>> if "goodbye" or "bye" in speech_text:
...     print("end")
...
end
```

Reply:
also what if goodbye or bye are not in speech_text? then the loop will never end right? But I guess that's where `KeyboardInterrupt` comes in? User has to manually break the script then?

Reply:
Good catch! Yeah I just came across this situation where "by" was detected, and the loop continued. Hmm. Wonder if a better way to break the loop vs just a KeyboardInterrupt. Lemme know what you think.

Reply:
So "by" not being transcribed as "bye" hence the if did not hit? Of course you cannot include "by" here because it's a too common word. So the issue is that the word "bye" is not always picked up from audio and/or transcribed well?